echo-artifacts/src/main/java/com/netflix/spinnaker/echo/config/ArtifactEmitterConfig.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.config;
+
+import com.jakewharton.retrofit.Ok3Client;
+import com.netflix.spinnaker.echo.events.RestClientFactory;
+import com.netflix.spinnaker.echo.rest.RestService;
+import com.netflix.spinnaker.retrofit.Slf4jRetrofitLogger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import retrofit.RestAdapter;
+import retrofit.converter.JacksonConverter;
+
+import static retrofit.Endpoints.newFixedEndpoint;
+
+@ComponentScan("com.netflix.spinnaker.echo.artifacts")
+@Configuration
+@EnableConfigurationProperties(ArtifactEmitterProperties.class)
+public class ArtifactEmitterConfig {
+
+  @Bean
+  public RestAdapter.LogLevel retrofitLogLevel(@Value("${retrofit.logLevel:BASIC}") String retrofitLogLevel) {
+    return RestAdapter.LogLevel.valueOf(retrofitLogLevel);
+  }
+
+  @Bean
+  @ConditionalOnExpression("${artifact-emitter.enabled:false}")
+  public ArtifactEmitterUrls artifactRestServices(
+    ArtifactEmitterProperties artifactEmitterProperties,
+    RestClientFactory clientFactory,
+    Ok3Client ok3Client,
+    RestAdapter.LogLevel retrofitLogLevel
+  ) {
+    ArtifactEmitterUrls artifactEmitterUrls = new ArtifactEmitterUrls();
+
+    for (ArtifactEndpointConfiguration endpoint : artifactEmitterProperties.endpoints) {
+
+      RestAdapter.Builder restAdapterBuilder = new RestAdapter.Builder()
+        .setEndpoint(newFixedEndpoint(endpoint.getUrl()))
+        .setClient(endpoint.internal ? ok3Client : clientFactory.getClient(endpoint.insecure))
```

Comment:
Why use a different client for internal and external requests?

Reply:
Internal clients are configured with some security-ness (can you tell I'm not an expert in this? ha ha). With the generic client factory we don't have the credentials to talk to our services. The thought is "internal=true" means we're sending this artifact to something like `keel` or `front50`. External would be like a separate auditing system or some service that's not in spinnaker.

Does that make sense? @ajordens maybe you could weigh in on if this is the right strategy or if I can use just one of the clients for everything.

Reply:
never mind @ajordens, I changed this to create a keel service and send events to that.